src/main/java/de/metas/ui/web/process/descriptor/InternalName.java

```diff
@@ -0,0 +1,84 @@
+package de.metas.ui.web.process.descriptor;
+
+import javax.annotation.Nullable;
+
+import org.adempiere.exceptions.AdempiereException;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonValue;
+
+import de.metas.util.Check;
+import lombok.EqualsAndHashCode;
+import lombok.NonNull;
+
+/*
+ * #%L
+ * metasfresh-webui-api
+ * %%
+ * Copyright (C) 2019 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ * 
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ * 
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+@EqualsAndHashCode(doNotUseGetters = true)
+public final class InternalName
+{
+	public static InternalName ofNullableString(@Nullable final String str)
+	{
+		return Check.isEmpty(str, true) ? null : ofString(str);
+	}
+
+	@JsonCreator
+	public static InternalName ofString(@NonNull final String str)
+	{
+		final String strNorm = normalizeString(str);
+		if (str.isEmpty())
+		{
+			throw new AdempiereException("Invalid internal name: " + str);
+		}
+
+		return new InternalName(strNorm);
+	}
+
+	private static String normalizeString(@NonNull final String str)
+	{
+		return str.trim()
+				.replace(" ", ""); // remove spaces because of cypress
```

Comment:
Would it look cleaner with underscores instead of spaces?

for `This is Something ACRONYM Bla` 
`This_is_Something_ACRONYM_Bla` vs `ThisisSomethingACRONYMBla`?

Reply:
sure, it's up to you guys.... i just wanted to provide a prototype

Reply:
in `normalizeString` method u can implement however looks better and is cypress friendly